### PR TITLE
feat(LoadingIndicator): add loader indicator component

### PR DESCRIPTION
**Pull request by booc0mtaco**
### Summary:

Add in a loader component, which can be used when there's a non-finite amount of time until loading is completed. Allow for some preliminary props to control size, and apply color overrides by CSS (blocking the component interface).

use the [react loader spinner](https://mhnpd.github.io/react-loader-spinner/docs/components/oval)'s `Oval` implementation. 

Future use case, apply an `as` prop to allow any other loader types using the same component (for example).

### Test Plan:

- [x] storybook
- [x] unit tests (for coverage)

### CODE REVIEW

File: src/components/Loader/Loader.tsx
Comment (jeremiah-clothier) [inline src/components/Loader/Loader.tsx:8]: This seems fine because we can also document the props ourselves with comments like you've done. Also, the types for `ariaLabel` or `visible` are straightforward

**Review by jeremiah-clothier — DISMISSED** [review]
I'm surprised chromatic handles the animation consistently and doesn't flake

File: src/index.ts
Comment (jinlee93) [inline src/index.ts:81]: Nice lol

**Comment by jinlee93** [comment]
Can I ask why this library for spinners over the others e.g. `react-spinners`?

**Comment by jinlee93** [comment]
The small seems very err small for lack of a better word.
image
Are there any accessibility requirements for loaders?

File: src/components/Loader/Loader.stories.ts
Comment (jinlee93) [inline src/components/Loader/Loader.stories.ts:18]: nit: not needed 
```suggestion
```

File: src/components/Loader/Loader.tsx
Comment (jinlee93) [inline src/components/Loader/Loader.tsx:56]: Think we generally include a short blurb about the component here in the docstring

File: src/components/Loader/Loader.tsx
Comment (jinlee93) [inline src/components/Loader/Loader.tsx:24]: Unfortunately these jsdoc tags are not super compatible with Storybook, so these urls are not visible in the docs tab of the story page.
image

File: src/index.ts
Comment (jinlee93) [reply src/index.ts:81]: Btw our plop command will generate recommended files and insert this line for you i.e. `yarn plop Loader` or `yarn plop Select`

Comment (booc0mtaco) [reply src/index.ts:81]: lolz, yeah. For that select component I made a copy of the Dropdown directory, so bypassed plop in that case

File: src/components/Loader/Loader.tsx
Comment (booc0mtaco) [reply src/components/Loader/Loader.tsx:56]: Yah forgot to add something as a little starter. We will have some revisions to this once Sean gets a design up for comparison.

File: src/components/Loader/Loader.tsx
Comment (booc0mtaco) [reply src/components/Loader/Loader.tsx:24]: Sad. While they won't show in Storybook (yet?), they do show up in the IDE, so an engineer can see the doc.s and have an active link while doing their work. 

Screen Shot 2022-12-08 at 11 35 50

File: src/components/Loader/Loader.tsx
Comment (booc0mtaco) [reply src/components/Loader/Loader.tsx:8]: yah, we can always revisit this in case there's a much better way to pluck them from the spinner type directly. I did try to do a type union, but there's some issue with the lib since it doesn't export the interface for `OvalProps` (plus, we don't want to export all the props anyhow... this component is a subset that may change in the future)

File: src/components/Loader/Loader.tsx
Comment (booc0mtaco) [inline src/components/Loader/Loader.tsx:18]: I am thinking we may want to have some EDS utility types for stuff like this. Codify all the well-known props so they don't vary unexpectedly. This one in particular might not be ideal tho, as not every component has all three sizes. Food for thought 🍔

**Comment by booc0mtaco** [comment]
@jinlee93 

* This library had a good variety of options AND was a bit smaller when unpacked, so went with it. When comparing the two components that had a similar style, this one came with a good set of props to cover the use cases Sean and I discussed as well. If we ever decide we don't want any of the other spinner types, we should probably pull out the SVG from this lib, as that extra bloat won't ever be needed. Lastly, `react-spinners` uses regular DOM elements `<span>` and CSS inline animations, while this one uses SVGs with relevant animation tags so it's a bit more self-contained.
* I made the sizes a little small on purpose to prompt a discussion on what the stroke sizes should be. Since the design is still pending, we can hold this PR until that is confirmed OR merge it in, then modify to suit. a11y-wise, we do have the label on this particular thing (and we should likely be using an `aria-live` somewhere when this component is rendered versus dismissed, but that's outside the scope of this one)

@jeremiah-clothier 

* I was curious about that too, so made a few pushes to see if the snapshots were stable. I think they are b/c the animation is triggered inside the SVG instead of via JS or thru CSS animations. Were it CSS, it might behave different based on the test page being `ready`, and JS obvs would be flakey since it might fire ... or not.

File: src/components/Loader/Loader.tsx
Comment (jinlee93) [reply src/components/Loader/Loader.tsx:23]: Makes sense to me, thanks for addressing

**Review by jinlee93 — APPROVED** [review]
Looks good pending mock updates and stroke width / general approval from design / Sean

File: src/components/Loader/Loader.tsx
Comment (dierat) [inline src/components/Loader/Loader.tsx:58]: `Loader` seems really vague to me and could mean several things (like a component that fetches data); I'd really prefer something like `LoadingSpinner`.

**Comment by dierat** [comment]
What's the screen reader experience like?

File: src/components/Loader/Loader.tsx
Comment (booc0mtaco) [reply src/components/Loader/Loader.tsx:58]: @dierat fair point. I have used `LoadingIndicator` in the past, which is less ambiguous and much less likely to conflict with other stuff.

I'm waiting for a figma before merging this so once we go thru that, I will make the naming update

**Comment by booc0mtaco** [comment]
> What's the screen reader experience like?

I am awaiting a POV from design on this one. I can imagine a few possible scenarios: the indicators are `aria-hidden` and designs include some text that gets read out, or the current `aria-label` exists as the sole representation of what is happening in the UI. Maybe some other options?

Related: there is probably a better default `aria-label` to apply than what the base component outputs (e.g., change the default to "Loading, please wait." or similar)

**Review by jeremiah-clothier — APPROVED** [review]

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (dierat) [inline src/components/LoadingIndicator/LoadingIndicator.tsx:20]: Can we use a different default aria label? "oval loading" sounds pretty weird. Just "loading" would be better.

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (dierat) [inline src/components/LoadingIndicator/LoadingIndicator.tsx:23]: Do we need the `@see` jsdoc tag here? It'd be nice to have this url show up on the storybook docs page.

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (dierat) [inline src/components/LoadingIndicator/LoadingIndicator.tsx:36]: I don't understand what the `(pixel = rem * 16)` part means. Are you saying a rem is 16 pixels?

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (booc0mtaco) [reply src/components/LoadingIndicator/LoadingIndicator.tsx:20]: Yah, certainly. This default comes from the library itself, but doesn't make much sense

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (booc0mtaco) [reply src/components/LoadingIndicator/LoadingIndicator.tsx:23]: Interesting, i didn't realize the `@see` was causing it to be hidden by storybook. If i remove that jsdoc directive, it will show up in storybook? I can test that.

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (booc0mtaco) [reply src/components/LoadingIndicator/LoadingIndicator.tsx:36]: Yep that's it. I can actually remove that and let the size comments tell people where to find the value

File: src/components/LoadingIndicator/LoadingIndicator.tsx
Comment (booc0mtaco) [reply src/components/LoadingIndicator/LoadingIndicator.tsx:23]: Screen Shot 2022-12-16 at 11 04 13

Comment (dierat) [reply src/components/LoadingIndicator/LoadingIndicator.tsx]: Yeah for some reason storybook just cannot handle those `@` tags and won't render it or anything after it 🤷

**Comment by jinlee93** [comment]
@booc0mtaco Just saw that we had a spinner [Icon](https://chanzuckerberg.github.io/edu-design-system/?path=/story/atoms-icons-icon--medium&args=name:spinner) and is being used in the `loading` variant of the [`Button`](https://chanzuckerberg.github.io/edu-design-system/?path=/story/molecules-buttons-button--loading). However, a quick regex search thru LP suggests this variant `<Button[\S\s*]*loading` isn't being used.

**Pull request merged**
